### ISSUE TITLE
Thread safety issue?

### ISSUE THREAD

**Issue by dmazin-drw** [sapcc/redfish-exporter#9]
Hi there,

If we issue a bunch of requests for different servers like so:
```bash
hosts=(server1 server2)
for host in "${hosts[@]}"; do
    curl -s "http://${RF_HOST}/health?target=$host-idrac&job=redfish" &
done
```

(note the & at the end of the curl command)

Then we run into an issue where the collector actually only talks to one of the hosts, regardless of the hostname.

I think this is because of a thread safety issue in handler.py.
Even though the app is multithreaded, the same instance of `metricsHandler` is used for all requests (this is correct behavior).
However, `on_get` function references `self`, which means that the requests clobber each other's values. 

Specifically, this is a problem for the code that translates hostnames to IPs and vice-versa.
Removing references to `self` in on_get fixes the issue.

Can you please let me know if I am off base here, or if this makes sense to you?

**Comment by BerndKue** [comment]
You are totally right. I also realized this after debugging for a long time and this is fixed by now.
